src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -3007,126 +3223,401 @@ static void Main()
+                //         F(() => 1, () => 2, string.Empty);
+                Diagnostic(ErrorCode.ERR_AmbigCall, "F").WithArguments("Program.F(System.Delegate, System.Delegate, string)", "Program.F(System.Func<int>, System.Delegate, MyString)").WithLocation(12, 9)
             };
-
-            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
-            comp.VerifyDiagnostics(expectedDiagnostics10AndLater);
-
+            comp = CreateCompilation(source, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics(expectedDiagnostics);
             comp = CreateCompilation(source);
-            comp.VerifyDiagnostics(expectedDiagnostics10AndLater);
+            comp.VerifyDiagnostics(expectedDiagnostics);
         }
 
         [Fact]
-        public void OverloadResolution_23()
+        public void OverloadResolution_39()
         {
             var source =
 @"using System;
-class Program
+using System.Linq.Expressions;
+class C
 {
-    static void F(Delegate d) => Console.WriteLine(""F(Delegate d)"");
-    static void F(Func<object> f) => Console.WriteLine(""F(Func<int> f)"");
+    static void M(Expression e) { Console.WriteLine(""M(Expression e)""); }
+    static void M(object o) { Console.WriteLine(""M(object o)""); }
+    static int F() => 0;
     static void Main()
     {
-        F(() => 1);
+        M(F);
+        M(() => 1);
     }
 }";
 
-            string expectedOutput = "F(Func<int> f)";
-            CompileAndVerify(source, parseOptions: TestOptions.Regular9, expectedOutput: expectedOutput);
-            CompileAndVerify(source, parseOptions: TestOptions.Regular10, expectedOutput: expectedOutput);
-            CompileAndVerify(source, expectedOutput: expectedOutput);
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (10,11): error CS1503: Argument 1: cannot convert from 'method group' to 'Expression'
+                //         M(F);
+                Diagnostic(ErrorCode.ERR_BadArgType, "F").WithArguments("1", "method group", "System.Linq.Expressions.Expression").WithLocation(10, 11),
+                // (11,11): error CS1660: Cannot convert lambda expression to type 'Expression' because it is not a delegate type
+                //         M(() => 1);
+                Diagnostic(ErrorCode.ERR_AnonMethToNonDel, "() => 1").WithArguments("lambda expression", "System.Linq.Expressions.Expression").WithLocation(11, 11));
+
+            var expectedDiagnostics = new[]
+            {
+                // (10,11): error CS0428: Cannot convert method group 'F' to non-delegate type 'Expression'. Did you intend to invoke the method?
+                //         M(F);
+                Diagnostic(ErrorCode.ERR_MethGrpToNonDel, "F").WithArguments("F", "System.Linq.Expressions.Expression").WithLocation(10, 11)
```

Comment:
Ideally, overload resolution should choose `M(object)` rather than `M(Expression)`.